### ISSUE TITLE
Terminating Conditions and Interface

### ISSUE THREAD

**Issue by TrentHawkins** [TrentHawkins/Chess#21]
Conditions by which the game ends:
- [x] Endgame:
    - [x] by checkmate,
    - [x] by resignation.
- [x] Draw:
    - [x] by stalemate,
    - by threefold round repetition (optional),
    - [x] by mutual agreement.
The game interface includes:
- [x] Movement reading:
    - [x] Chess long [algebraic move notation](https://en.wikipedia.org/wiki/Algebraic_notation_(chess)) reading
    - FIDE short notation, which is hard to parse with lots of special cases (optional).
- [x] Game loop:
    - [x] Advancing turns:
        - [x] Read input from players.
        - [x] Keep track of player stats.
            - [x] Player pieces.
            - [x] Captured pieces.
            - [x] Display said stats on main screen
- [x] Various visual improvements to the interface.

**Comment by TrentHawkins** [comment]
Interface stuff moved here because they are implemented in parallel to terminating conditions.

**Comment by TrentHawkins** [comment]
# Optional features
All done and tested. Game terminates on all but the three-fold condition. I address the two optional features I have not implemented, and why they are low priority:

From these two features, I'd say the latter takes precedence in implementation, even if a graphical interface is added to the game later (mouse and click), for reasons of documenting the move history with short notation.

## Three-fold repetition
I consider the three-fold repetition draw condition a low priority, as players can simply mutually agree to a draw when that happens. While stalemate as a draw is important because it's a trap a player can fall into by miscalculation. Three-fold repetition on the other hand is always deliberate, and on the off-chance it happens by accident with neither player intending to draw the game, it would be a pity to lose the game.

## Short notation
FIDE's short notation is the standard, however it involves checks beyond advanced regex-ing. Move long notation is abbreviated if there is no confusion with relation to the board state. Therefore, only piece type and target square are annotated unless, more than one pieces per player potentially target that square, so the move notation adds the source file the piece is moving from. If the file of the conflicting pieces is the same, use the source rank instead. In the rare occasion that both source file and rank conflict (this can happen when three pieces of the same type target the same square, with a pair having the same file and another pair having the same rank), use the full source square (this move's annotation will then in effect match the long notation used in the current implementation). These checks constitute cumbersome code that will require off-line method updating of `Move` objects and derivatives, the same way `Piece` objects are, because the short-handing depends on board context.

The bright-side is that context needed stops at the board level, however they still need to be set at a player's level, as their piece collection is perused.

# Remarks on #22
I should comment that this PR #22 has also fixed several bugs to the previous (merged) one, even though the previous one #17 was operational at 90% of the cases it was to cover. I should note that the discovery of the few bugs in #17 was only possible with the work put on #22, which means leaving #17 open would require putting a lot of work out of its scope to validate it fully. If, on the other hand, I would (as I did actually) only branch out from my work locally, I would be unable to send you updates on what is now PR #22. So please revise this super-working #22 and consider merging it.
